ext/sockets/sockets.c

```diff
@@ -884,7 +884,7 @@ PHP_FUNCTION(socket_read)
 	ENSURE_SOCKET_VALID(php_sock);
 
 	/* overflow check */
-	if ((length + 1) < 2) {
+	if (length <= 0 || length == ZEND_LONG_MAX) {
```

Comment:
Actually, the max allowed string length is ZSTR_MAX_LEN, which is less than ZEND_LONG_MAX.
So you should change the checks to
```suggestion
	if (length <= 0 || length > ZSTR_MAX_LEN) {
```

Reply:
Similarly, the `length` check in `socket_recvfrom` should be adjusted.